internal/featuretests/kubernetes.go

```diff
@@ -32,77 +35,74 @@ func IngressBackend(svc *v1.Service) *networking_v1.IngressBackend {
-ejhNdEECgYEA/3shOSl4GmEGZZXTARo0bDnelSpSJgmlStBeAGraq+lm/4IrTCCB
-2tmJkEs6uiCEH7vkbbI/lLlhNjM2gKDovP3UwTQHGfiHlcIYYICMIpbp7+Ar9+ex
-4KPXmgqqhTKC4xQYGoUE6INlpZcQ4blA4AtQbhgcC4Cp8FD/QbXp0xECgYEA02Ll
-EAxzHZBNoK/5oPL0LiDUW/zPdOWCQCW8oGW5gDUvCLQ/lntegL8Qzubt7PZ4xgeg
-m2ENTDcp1Zfn9s0T1V2T9Sba8gShUCvm9nLVCj4OQ3lwDufwHFuhKFpj1BVnHD5y
-9yhXfyrFgvhamepEjq6LZUCrL1HxZqgOezv6JccCgYBqM1oFNArcFFcfZV+YRrdi
-AdBX+4a4jyvp5KIe1ExgSB7rucWb2KuCOQmpNMyN0LR7qJR1UTKC9WjGqhVO9RSq
-c228fo8xKZHbHBscCnO2cTt/3pUIcYUM167pNuPZiLzF/nVimMcIjI51fk2jN2oT
-eECP82+9DFgYMONbAm7XsQKBgH/Y3ztenDz0Ks8Vv3/FkUNY3bco5vwHV0ieyj+k
-ZpYRFHpKMe88fEKXzH2mk53uz8rNkCiJgTZoYqfpcQUGsYkpSLRLpL4daMcJVm4V
-s523PH84sjqBsuojzQuP57K8oxkk9/ld79VctAprVLikRISbMnmxrBc5kywIVoHY
-G4m/AoGBAPV0wuytURR3CbPHz28wrKRh/xnHrW+Fp3ooRfj8Pr/4zDVsqqNz2gA3
-yVb6kAEpg2ON9NOWSfoUfC+THitOnRm8pKL1QL7oiq/2+s3IiK+jSevEU7TUfjio
-1LwtUqv1MbKdv7TgkU0YQ99iLocWF4F4oWF6AX86/BL9y7gcbE0y
------END RSA PRIVATE KEY-----`
-
-	CRL = `-----BEGIN X509 CRL-----
-MIHlMIGMAgEBMAoGCCqGSM49BAMCMBsxGTAXBgNVBAMTEGNsaWVudC1yb290LWNh
-LTEXDTIyMDYyMTA5NDQ0NVoXDTIyMDYyODA5NDQ0NVowGzAZAggW+pmWu/XnExcN
-MjIwNjIxMDk0NDQ1WqAjMCEwHwYDVR0jBBgwFoAULRlmjBtfjbzwV2WeO9Vj5pWO
-h5gwCgYIKoZIzj0EAwIDSAAwRQIhANVFCqByuASAcbz6ovyvi5KCtPfNjHjxVaNT
-x69LFPN1AiA5pF5rqHy1FBctZBTW+3LTEEX35j3p1++zcNu8oHMO/w==
------END X509 CRL-----`
-)
+var CACertificate = certyaml.Certificate{
+	Subject: "CN=ca",
+}
+
+var ServerCertificate = certyaml.Certificate{
+	Issuer:          &CACertificate,
+	Subject:         "CN=www.example.com",
+	SubjectAltNames: []string{"DNS:www.example.com"},
+}
+
+var ClientCertificate = certyaml.Certificate{
+	Issuer:  &CACertificate,
+	Subject: "CN=client",
+}
 
-func Secretdata(cert, key string) map[string][]byte {
-	return map[string][]byte{
-		v1.TLSCertKey:       []byte(cert),
-		v1.TLSPrivateKeyKey: []byte(key),
+var CRL = certyaml.CRL{
+	Issuer: &CACertificate,
+}
+
+func TLSSecret(name string, credential *certyaml.Certificate) *v1.Secret {
+	cert, key, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeTLS,
+		Data: map[string][]byte{
+			v1.TLSCertKey:       cert,
+			v1.TLSPrivateKeyKey: key,
+		},
 	}
 }
 
+func CASecret(name string, credential *certyaml.Certificate) *v1.Secret {
+	cert, _, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			dag.CACertificateKey: cert,
+		},
+	}
+}
+
+func CRLSecret(name string, credential *certyaml.CRL) *v1.Secret {
+	crl, err := credential.PEM()
+	if err != nil {
+		panic(err)
+	}
+
+	return &v1.Secret{
+		ObjectMeta: fixture.ObjectMeta(name),
+		Type:       v1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			dag.CRLKey: crl,
+		},
+	}
+}
+
+func PEMBytes(cert *certyaml.Certificate) []byte {
+	c, _, _ := cert.PEM()
```

Comment:
here we're ignoring the error, but in other places we use a `panic`

should we either ignore the error everywhere or maybe make these test helpers and pass `testing.T` so we can make an assertion there is no error?

Reply:
Done!